### DISCUSSION TITLE
A NodeJS major version upgrade updates only Dockerfile and not .nvmrc unlike Nodejs minor upgrade

### DISCUSSION THREAD

**Discussion by pre** [renovatebot/renovate#22127] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.73.0

### If you're self-hosting Renovate, select which platform you are using.

None

### If you're self-hosting Renovate, tell us what version of the platform you run.

N/A

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

A NodeJS major version upgrade creates a different kind of a PR than a minor version update:

Major: Updates only `Dockerfile` , does not update `.nvmrc`
* PR title: "Update Node.js to v20"

Minor: Updates both `Dockerfile` and `.nvmrc`
* PR title: "Update Node.js to v18.16.0"

Expected: Major Node.js PR updates both `Dockerfile` and `.nvmrc`

### What you tried so far.

Renovate creates PRs as expected, but the major version upgrades act differently than minor.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by pre** [comment]
Extracted from an auto-closed PR https://github.com/renovatebot/renovate/issues/22125

**Reply by rarkins** [reply]
> Extracted from an auto-closed PR #22125

![image](https://github.com/renovatebot/renovate/assets/6311784/dda8d270-8f53-4e99-a7e8-36db8f5d7f05)

![image](https://github.com/renovatebot/renovate/assets/6311784/8a656e3b-2844-483f-bdb1-f6874b5f978e)

**Comment by rarkins** [comment, marked as answer]
It's Renovate's default/intended behavior to not propose updates from Node v18 to Node v20 by default, as that is going from LTS to non-LTS. However, for some Docker images this capability is not supported if the image uses suffixes like `node:18-alpine`.

In other words: suppressing v20 from nvmrc is a feature which doesn't exist yet for all Docker tag formats

**Reply by pre** [reply]
Thanks for the answer! Here's the context:

PR for Major version is `20.1.0` which could be set in `.nvmrc` as is:
<img width="579" alt="image" src="https://github.com/renovatebot/renovate/assets/63895/c6dc6849-7763-4e8d-a144-ee31969749ae">

PR for Minor version:
<img width="504" alt="image" src="https://github.com/renovatebot/renovate/assets/63895/fc21e5f7-8199-4188-9dea-369fb49360d7">

So the Major version tag is not a general "20" but instead a specific `20.1.0`. Is there a way to wiggle this into both Dockerfile and .nvmrc automatically?

**Reply by rarkins** [reply]
I _thought_ that we automatically set node versioning for node images with semver tags, but your screenshot indicates otherwise and I can't find any source code to dispute it.

If you configure a package rule with matchDatasources=docker, matchPackageNames=node, and versioning=node, then it would try to use node versioning for anything named `node`. If you use any `-something` suffixes (such as `-alpine`) then you'll need a more specific rule to avoid it
